### DISCUSSION TITLE
Problem in Select DropDown Menu

### DISCUSSION THREAD

**Discussion by hanan-commits-work** [iTwin/itwinjs-core#3111] [Q&A]
Greetings.

I am having problems in dealing with the UI of select following the 3.0 update. 

I am using select from itwinui-react, without any styling (MenuStyle, menuItemStyle). The behavior I am expecting here, is that select dropdowns a menu precisely underneath it, (or above it in case that select is positioned at the bottom of the screen).
But i am experiencing rather interesting bugs when I am making a select inside a dialog box. let me explain it with screenshots,

In this i clicked on the "Comment" select box, but the dropdown menu hasnt dropped, (I believe that the dropdown menu is placed outside the screen).
![image](https://user-images.githubusercontent.com/89520756/151129276-d02ede31-a5dc-4e08-b4ec-f48249cf2f3f.png)

 But if I move the dialog box and then I press the select menu (comment or priority), the dropdown appears at the correct location and is working properly. As you will see in the screenshot below.
![image](https://user-images.githubusercontent.com/89520756/151129680-d096d410-658d-4986-b7ce-422559d0c049.png)

I am uploading the sample project featuring these issues.
Here is the link,
https://github.com/hanan-commits-work/itwins3.0-guidance

Please install the project, fill the env file with your variables, and then run it. You will see DropDown Button widget in the side menu,
1. Click on the button and then dialog will appear, 
2. after that click on either of the dropdowns
3. The dropdown wouldnt appear, (which is the bug)
4. Then slightly move the dialog box and then click on the dropdowns and they will appear to be functioning properly.

For debugging purposes, the file you need to refer to will be uiProvider.tsx 

Here is the line for select, 
For the select option in dialog box:
https://github.com/hanan-commits-work/itwins3.0-guidance/blob/214fd5d738dcf59eb3aa5784984d06a3d203fc81/src/uiProvider.tsx#L36

Any help would be appreciated.

**Comment by bsteinbk** [comment]
@veekeys @bentleyvk

**Comment by hanan-commits-work** [comment]
Any followup on this?

**Reply by bentleyvk** [reply]
Hey @hanan-commits-work, we are currently investigating this issue. As a temporary workaround you can add `popoverProps={{ 
appendTo: document.body }}`

**Reply by bentleyvk** [reply]
I think I was not clear with a workaround. You need do like this:
```tsx
<Select
  // your other Select props
  popoverProps={{ appendTo: document.body }}
/>
```
Let us know if it worked for you 🙂

**Reply by hanan-commits-work** [reply]
Hi, It didnt work. 
I have committed the change in the code so you can see.

**Reply by bentleyvk** [reply]
Sorry for not responding for so long. I tried this on your `UiProvider` and it worked:
```tsx
<Select
  // your other Select props
  popoverProps={{ appendTo: document.body, zIndex: 99999 }}
/>
```
We will investigate the possibility of setting some default values so you do not need to apply these in the future.
